docs/ugcstor-csi.md

```diff
@@ -0,0 +1,802 @@
+  Normal   FileSystemResizeRequired    88s                external-resizer cstor.csi.openebs.io                                                     Require file system resize of volume on node
+  Normal   FileSystemResizeSuccessful  4s                 kubelet, 127.0.0.1                                                                        MountVolume.NodeExpandVolume succeeded for volume "pvc-849bd646-6d3f-4a87-909e-2416d4e00904"
+```
+
+```
+$ kubectl get pvc
+```
+
+Sample Output:
+```
+NAME                           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS              AGE
+cstor-pvc                      Bound    pvc-849bd646-6d3f-4a87-909e-2416d4e00904   10Gi        RWO            cstor-csi-disk            1d
+```
+
+```
+$ kubectl get pv
+```
+
+Sample Output:
+
+```
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                   STORAGECLASS        REASON   AGE
+pvc-849bd646-6d3f-4a87-909e-2416d4e00904   10Gi        RWO            Delete           Bound    default/cstor-pvc       cstor-csi-disk               40m
+```
+<hr>
+
+### <a class="anchor" aria-hidden="true" id="performance-tunings-in-cstor-pools"></a>Performance Tunings in cStor Pools
+
+Allow users to set available performance tunings in cStor Pools based on their workload. cStor pool(s) can be tuned via CSPC and is the recommended way to do it. Below are the tunings that can be applied:
+<br>
+ 
+**Resource requests and limits:** This ensures high quality of service when applied for the pool manager containers.  
+
+**Toleration for pool manager pod:** This ensures scheduling of pool pods on the tainted nodes.
+
+**Set priority class:** Sets the priority levels as required.
+
+**Compression:** This helps in setting the compression for cStor pools.
+
+**ReadOnly threshold:** Helps in specifying read only thresholds for cStor pools.
+<br>
+
+<font size="4">**Example configuration for Resource and Limits:**</font>
+
+Following CSPC YAML specifies resources and auxResources that will get applied to all pool manager pods for the CSPC. Resources get applied to cstor-pool containers and auxResources gets applied to sidecar containers i.e. cstor-pool-mgmt and pool-exporter.
+<br>
+In the following CSPC YAML we have only one pool spec (@spec.pools). It is also possible to override the resource and limit value for a specific pool. 
+
+```
+apiVersion: cstor.openebs.io/v1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-disk-pool
+  namespace: openebs
+spec:
+  resources:
+    requests:
+      memory: "2Gi"
+      cpu: "250m"
+    limits:
+      memory: "4Gi"
+      cpu: "500m"
+
+  auxResources:
+    requests:
+      memory: "500Mi"
+      cpu: "100m"
+    limits:
+      memory: "1Gi"
+      cpu: "200m"
+  pools:
+    - nodeSelector:
+        kubernetes.io/hostname: worker-node-1
+
+      dataRaidGroups:
+      - cspiBlockDevices:
```

Comment:
`cspiBlockDevices` is not in the schema that we follow... also with several other nitpicks, it should look something like this:
```yaml
apiVersion: cstor.openebs.io/v1
kind: CStorPoolCluster
metadata:
  name: cstor-disk-pool
  namespace: openebs
spec:
  resources:
    requests:
      memory: "64Mi"
      cpu: "250m"
    limits:
      memory: "128Mi"
      cpu: "500m"
  auxResources:
    requests:
      memory: "50Mi"
      cpu: "400m"
    limits:
      memory: "100Mi"
      cpu: "400m"
  
  pools:
    - nodeSelector:
        kubernetes.io/hostname: worker-node-1
      dataRaidGroups:
        - blockDevices:
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f36
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f37
      poolConfig:
        dataRaidGroupType: mirror
    
    - nodeSelector:
        kubernetes.io/hostname: worker-node-2
      dataRaidGroups:
        - blockDevices:
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f39
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f40
      poolConfig:
        dataRaidGroupType: mirror
    
    - nodeSelector:
        kubernetes.io/hostname: worker-node-3
      dataRaidGroups:
        - blockDevices:
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f42
            - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f43
      poolConfig:
        dataRaidGroupType: mirror
        resources:
          requests:
            memory: 70Mi
            cpu: 300m
          limits:
            memory: 130Mi
            cpu: 600m
        auxResources:
          requests:
            memory: 60Mi
            cpu: 500m
          limits:
            memory: 120Mi
            cpu: 500m
```